### PR TITLE
refactor: flakey test and refactor

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/BuildingRegistry/Building/Building_BuildingUnit.cs [hunk 1]
```diff
@@ -53,32 +54,18 @@ public void RealizeBuildingUnit(BuildingUnitPersistentLocalId buildingUnitPersis
...
-            if (buildingUnit is null)
-            {
-                throw new BuildingUnitIsNotFoundException(
-                    BuildingPersistentLocalId,
-                    buildingUnitPersistentLocalId);
-            }
-
-            buildingUnit.Realize();
+            _buildingUnits
+                .GetNotRemovedByPersistentLocalId(buildingUnitPersistentLocalId)
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building_BuildingUnit.cs:58]: @ArneD doubting if we want to keep or delete the GuardRemoved() on BuildingUnit methods (Realize, NotRealize ...).

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/BuildingRegistry/Building/Building_BuildingUnit.cs (hunk 1 above)
Comment (ArneD) [reply src/BuildingRegistry/Building/Building_BuildingUnit.cs:58]: I'm conflicted. On the one hand you have an aggregate witch is the entry point of your domain and thus the checks are done once entering the domain.
On the other hand if you code in the domain and call `.Realize` I would expect that the `Entity` also protects itself from invalid state / actions.

Comment (jvandaal) [reply src/BuildingRegistry/Building/Building_BuildingUnit.cs:58]: Was having the same thing in mind. Let's for now keep it like that.

File: src/BuildingRegistry/Building/Building_BuildingUnit.cs [hunk 2]
```diff
@@ -228,7 +173,7 @@ public void RemoveBuildingUnit(BuildingUnitPersistentLocalId buildingUnitPersist
             NotRealizeOrRetireCommonBuildingUnit();
         }
 
-        private void GuardBuildingInvalidStatuses(BuildingStatus[] invalidStatuses)
+        private void GuardBuildingInvalidStatuses(IEnumerable<BuildingStatus> invalidStatuses)
```
[the remark this answers is not in the document]
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building_BuildingUnit.cs]: Changed.

**Review by ArneD — APPROVED** [review]

**Pull request merged**